src/main.rs

```diff
@@ -10,6 +10,38 @@ use libazureinit::{
 
 const VERSION: &str = env!("CARGO_PKG_VERSION");
 
+fn get_username(
+    imds_body: String,
+) -> Result<String, Box<dyn std::error::Error>> {
+    if imds::is_password_authentication_disabled(&imds_body).map_err(|_| {
+        "Failed to get disable password authentication".to_string()
+    })? {
+        // password authentication is disabled
+        match imds::get_username(imds_body.clone()) {
+            Ok(username) => Ok(username),
+            Err(_err) => Err("Failed to get username".into()),
+        }
```

Comment:
This was a case I missed that could also be a `map_err`, but it's up to you if you want to adjust it. I'm touching all the errors in #59 anyway and need to rebase it after this is merged so I can adjust it there.

Reply:
It looks a little tricky touch that in this PR, because `main` does not return error in this context.
It might be probably better to touch in your PR.